### PR TITLE
ID-16612: Adyen 3Ds security adaptations

### PR DESCRIPTION
**Pull request by TarasBozhok**
[ADYEN][SFCC] Security adaptations due to Adyen cartridge version (3DS)

### CODE REVIEW

File: cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js [hunk 1]
```diff
@@ -10,22 +10,33 @@ var Resource = require('dw/web/Resource');
...
+        IssuerURL.substr(IssuerURL.length - 25),
+        MD.substr(1, 25)
+    );
+
+    if (signature === currentSignature) {
+        res.render('adyenform', {
+          issuerUrl: IssuerURL,
+          paRequest: PaRequest,
+          md: MD,
+          ContinueURL: TermURL,
```
Comment (mchamponnois) [inline cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js:33]: indentation issue + the comma at the end will break the process (not supported with our compatibility mode on TAG Heuer), did you test your changes ?

File: cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js [hunk 2]
```diff
@@ -10,22 +10,33 @@ var Resource = require('dw/web/Resource');
...
+        });
+        return next();
+    }
+    Logger.getLogger('Adyen').error('Signature incorrect for 3DS payment');
+    res.redirect(URLUtils.url('Home-Show', 'Payment','Failed3DS'));
```
Comment (mchamponnois) [inline cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js:38]: eslint: always space after comma

File: cartridges/int_adyen_SFRA/cartridge/scripts/hooks/payment/processor/adyen_credit.js [hunk 3]
```diff
@@ -79,38 +79,54 @@ function Authorize(orderNumber, paymentInstrument, paymentProcessor) {
...
 
-        return {
-            ThreeDS2: result.ThreeDS2,
-            resultCode: result.resultCode,
-            token3ds2: result.token3ds2,
+        if (result.threeDS2) {
+            return {
+                ThreeDS2: result.ThreeDS2,
+                resultCode: result.resultCode,
+                token3ds2: result.token3ds2,
```
Comment (mchamponnois) [inline cartridges/int_adyen_SFRA/cartridge/scripts/hooks/payment/processor/adyen_credit.js:95]: remove the comma at the end

**Review by mchamponnois — CHANGES_REQUESTED** [review]

File: cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js (hunk 1 above)
Comment (TarasBozhok) [reply cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js:33]: Yes

Comment (TarasBozhok) [reply cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js]: Indentation fixed, comma removed.

File: cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js (hunk 2 above)
Comment (TarasBozhok) [reply cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js]: Space added.

File: cartridges/int_adyen_SFRA/cartridge/scripts/hooks/payment/processor/adyen_credit.js (hunk 3 above)
Comment (TarasBozhok) [reply cartridges/int_adyen_SFRA/cartridge/scripts/hooks/payment/processor/adyen_credit.js]: Comma removed.

**Review by mchamponnois — APPROVED** [review]

**Pull request merged**